output/cloud/output.go

```diff
@@ -383,24 +384,33 @@ func (out *Output) AddMetricSamples(sampleContainers []metrics.SampleContainer)
 				values[metrics.IterationsName] = 1
 			}
 
+			encodedTags, err := easyjson.Marshal(sc.GetTags())
+			if err != nil {
+				out.logger.Error("Encoding tags failed", err)
+			}
 			newSamples = append(newSamples, &Sample{
 				Type:   DataTypeMap,
 				Metric: "iter_li_all",
 				Data: &SampleDataMap{
 					Time:   toMicroSecond(sc.GetTime()),
-					Tags:   sc.GetTags(),
+					Tags:   encodedTags,
 					Values: values,
 				},
 			})
 		default:
 			for _, sample := range sampleContainer.GetSamples() {
+				encodedTags, err := easyjson.Marshal(sample.Tags)
+				if err != nil {
+					out.logger.Error("Encoding tags failed", err)
+				}
+
 				newSamples = append(newSamples, &Sample{
 					Type:   DataTypeSingle,
 					Metric: sample.Metric.Name,
 					Data: &SampleDataSingle{
 						Type:  sample.Metric.Type,
 						Time:  toMicroSecond(sample.Time),
-						Tags:  sample.Tags,
+						Tags:  encodedTags,
```

Comment:
Is this needed - I am pretty sure this will be done automatically ? Or does it not get done automatically?

Reply:
I've changed the type from `SampleTags` to `json.RawMessage` otherwise we can't use the unmarshal function, we have to unmarshal to a temporary type and manually resolve using the Atlas root.

Reply:
Unmarshalling the clout output metrics I think is only done in tests?